discovery/db_discovery.go

```diff
@@ -19,38 +18,52 @@ import (
 	"github.com/golang/glog"
 )
 
-var cacheRefreshInterval = 1 * time.Hour
-var getTicker = func() *time.Ticker {
-	return time.NewTicker(cacheRefreshInterval)
+type ticketParamsValidator interface {
+	ValidateTicketParams(ticketParams *pm.TicketParams) error
+}
+
+type roundsManager interface {
+	LastInitializedRound() *big.Int
 }
 
 type DBOrchestratorPoolCache struct {
-	node *core.LivepeerNode
+	store                 common.OrchestratorStore
+	lpEth                 eth.LivepeerEthClient
+	ticketParamsValidator ticketParamsValidator
+	rm                    roundsManager
 }
 
-func NewDBOrchestratorPoolCache(node *core.LivepeerNode) *DBOrchestratorPoolCache {
-	if node.Eth == nil {
-		glog.Error("Could not refresh DB list of orchestrators: LivepeerNode nil")
+func NewDBOrchestratorPoolCache(store common.OrchestratorStore, lpEth eth.LivepeerEthClient, rm roundsManager, ticketParamsValidator ticketParamsValidator) *DBOrchestratorPoolCache {
+	if store == nil {
```

Comment:
I don't think any of these params could be nil besides `lpEth` (in on-chain mode) so we could just have a single nil check for `lpEth`

Reply:
fixed in 74f273ef